### ISSUE TITLE
Convenience method for accessing individual merged cells, rather than just a reference

### ISSUE THREAD

**Issue by caramdache** [weshatheleopard/rubyXL#381]
Merged cells are a little bit of a red herring. Sometimes you want to access the actual cells rather than the reference/the MergedCell's. For example to obtain indivual borders for each of the cells in the merge. Which is required to be able to reproduce the exact borders.

In doing this work, I have found the following method useful. It may be a desirable inclusion in the library, probably under a more sophisticated form.

```ruby
def merged_cells(cell)
    return nil unless cell.worksheet.merged_cells
    
    cell.worksheet.merged_cells.each { |mcell|
        if mcell.ref.first_row == cell.row && mcell.ref.first_col == cell.column
            return mcell.ref.row_range.collect { |row|
                       mcell.ref.col_range.collect { |col|
                           cell.worksheet[row][col]
                       }
                   }
        end
    }
    
    nil
end
```

**Comment by caramdache** [comment]
Here's a more elaborate version.

```ruby
def borders(cell)
    [:top, :left, :right, :bottom].flat_map { |direction|
        merged_cells(cell).map { |c|
            direction if c.get_border(direction)
        }
    }.uniq
end

def merged_cells(cell)
    ws = cell.worksheet

    ws.merged_cells.each { |mcell|
        row_range, col_range = mcell.ref.row_range, mcell.ref.col_range

        if row_range.begin == cell.row && col_range.begin == cell.column
            return row_range.flat_map { |row| col_range.map { |col| ws[row][col] } }
        end
    } if ws.merged_cells
    
    [cell]
end
```

**Comment by harsh183** [comment]
Bumping this, I would definitely find it useful in one of my scripts.

**Comment by weshatheleopard** [comment]
@harsh183 Please propose the consumer facing API.

**Comment by harsh183** [comment]
Something like `sheet.merged_cell_at([row, col])` since we don't want to change any behavior of [][] access giving normal cells for even merged ones. This can return `MergedCell.ref()` would give us where we can access the range variables.

```ruby
sheet = parsed_file[0]
sheet.merged_cell_at([5, 3])
sheet.merged_cell_at(RubyXL::Reference.ref2ind('A1')) # why I'm thinking of array input
```

**Comment by weshatheleopard** [comment]
@harsh183 So you are saying that for given (X,Y) you want to know the (X1....X2, Y1.....Y2) of the merged cell, of which the cell (X,Y) is a part of?

**Comment by harsh183** [comment]
Yeah that works, and being able to get/set properties of the entire merged cell.

**Comment by caramdache** [comment]
> Yeah that works, and being able to get/set properties of the entire merged cell.

Some properties are not set at the global merged cell level, but at the level of the individual cells which are merged. Borders are one example. The API should allow support for that.

**Comment by weshatheleopard** [comment]
@caramdache precisely, so use your chance, go ahead and propose an API!

**Comment by harsh183** [comment]
@caramdache do you mean something like `merged_cell.first_cell` or maybe some iterator that gives all the individual cells that are part of a given merged cell for those you can't set at a global merged cell level? 

And the rest can have a merged cell level set for things that can be like content?

For my project I'm just looking for merged_cell_at(x,y) but this is a part of the gem that can really be expanded on in interesting directions I feel.

**Comment by caramdache** [comment]
@harsh183 I meant an iterator that provides the individual cells that are part of a merge range.

@weshatheleopard my use case was to be able to obtain the borders of a group of merged cells. The borders are  defined for each individual cell, hence the need for an iterator to obtain these cells.

**Comment by harsh183** [comment]
Yeah, I think that'll be generally useful for a lot of things.
